packages/material-renderers/src/layouts/MaterialCategorizationLayout.tsx

```diff
@@ -96,8 +96,12 @@ export const MaterialCategorizationLayoutRenderer = (props: MaterialCategorizati
   const categories = useMemo(() => categorization.elements.filter((category: Category) =>
     isVisible(category, data, undefined, ajv)
   ),[categorization, data, ajv]);
+
+  const safeCategory = activeCategory >= categorization.elements.length ? 0 : activeCategory;
+  useEffect(()=>{setActiveCategory(0)}, [uischema]);
```

Comment:
I think we don't need an effect here.
Instead, you could do
```typescript
if (safeCategory !== activeCategory) {
  setActiveCategory(safeCategory)
}
```
This also has the advantage of not updating the active category if it is not necessary. Avoiding an effect is also advantageous for performance because it safes a render cycle. See here for more details on avoiding effects if you are interested: https://react.dev/learn/you-might-not-need-an-effect

Reply:
Thank you for the input, This are some good tricks!
If the `Uischema` changes and the category is still in range, the `activeCategory `will not reset to 0, but stay on the previously selected tap. I used another method from your link to reset it to 0 without useEffect.